src/lib/helpers.ts

```diff
@@ -66,3 +66,60 @@ export function applySettingDefaults<
 
   return result
 }
+
+export const BASE64URL_REGEX = /^([a-z0-9_-]{4})*($|[a-z0-9_-]{3}$|[a-z0-9_-]{2}$)$/i
+
+/**
+ * Checks that the value somewhat looks like a JWT, does not do any additional parsing or verification.
+ */
+export function isJWT(value: string): boolean {
+  if (value.startsWith('Bearer ')) {
+    value = value.substring('Bearer '.length)
+  }
+
+  value = value.trim()
+
+  if (!value) {
+    return false
+  }
+
+  const parts = value.split('.')
+
+  if (parts.length !== 3) {
+    return false
+  }
+
+  for (let i = 0; i < parts.length; i += 1) {
+    const part = parts[i]
+
+    if (part.length < 4 || !BASE64URL_REGEX.test(part)) {
+      return false
+    }
+  }
+
+  return true
+}
+
+export function checkAuthorizationHeader(headers: { [header: string]: string }) {
+  if (headers.authorization && headers.Authorization) {
+    console.warn(
+      '@supabase-js: Both `authorization` and `Authorization` headers specified in createClient options. `Authorization` will be used.'
+    )
+
+    delete headers.authorization
+  }
+
+  const authorization = headers.Authorization ?? headers.authorization ?? null
+
+  if (!authorization) {
+    return
+  }
+
+  if (authorization.startsWith('Bearer ') && authorization.length > 'Bearer '.length) {
```

Comment:
> `authorization.length > 'Bearer '.length` 

not immediately obvious what this check is for?

Reply:
Looks like it's making sure that we only check for a valid JWT if the authorization header actually has a value after "Bearer ".

It's worth noting that this check will not cover a scenario where someone only passes in "Bearer" (without a trailing space), but I'm guessing this whole scenario would be extremely rare anyway. Still, you might consider covering both.